sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -0,0 +1,484 @@
+      case Some(certificate) => certificate.quality
+      case None => 0
+    }
+  }
+
+  override def hasCeased: Boolean = stateMetadataStorage.hasCeased
+
+  override def getFeePayments(withdrawalEpoch: Int, blockToAppendFeeInfo: Option[AccountBlockFeeInfo] = None): Seq[AccountPayment] = {
+    val feePaymentInfoSeq = stateMetadataStorage.getFeePayments(withdrawalEpoch)
+    AccountFeePaymentsUtils.getForgersRewards(feePaymentInfoSeq)
+  }
+
+  def getWithdrawalEpochInfo: WithdrawalEpochInfo = stateMetadataStorage.getWithdrawalEpochInfo
+
+  def getConsensusEpochNumber: Option[ConsensusEpochNumber] = stateMetadataStorage.getConsensusEpochNumber
+
+  def getOrderedForgingStakesInfoSeq: Seq[ForgingStakeInfo] = using(getView)(_.getOrderedForgingStakesInfoSeq)
+
+  // Returns lastBlockInEpoch and ConsensusEpochInfo for that epoch
+  // TODO this is common code with SidechainState
+  def getCurrentConsensusEpochInfo: (ModifierId, ConsensusEpochInfo) = {
+    val forgingStakes: Seq[ForgingStakeInfo] = getOrderedForgingStakesInfoSeq
+    if (forgingStakes.isEmpty) {
+      throw new IllegalStateException("ForgerStakes list can't be empty.")
+    }
+
+    getConsensusEpochNumber match {
+      case Some(consensusEpochNumber) =>
+        val lastBlockInEpoch = bytesToId(stateMetadataStorage.lastVersionId.get.data) // we use block id as version
+        val consensusEpochInfo = ConsensusEpochInfo(
+          consensusEpochNumber,
+          MerkleTree.createMerkleTree(forgingStakes.map(info => info.hash).asJava),
+          forgingStakes.map(_.stakeAmount).sum
+        )
+        (lastBlockInEpoch, consensusEpochInfo)
+      case _ =>
+        throw new IllegalStateException("Can't retrieve Consensus Epoch related info form StateStorage.")
+    }
+  }
+
+  // Account specific getters
+  override def getBalance(address: Array[Byte]): BigInteger = using(getView)(_.getBalance(address))
+
+  override def getAccountStateRoot: Array[Byte] = stateMetadataStorage.getAccountStateRoot
+
+  override def getCodeHash(address: Array[Byte]): Array[Byte] = using(getView)(_.getCodeHash(address))
+
+  override def getNonce(address: Array[Byte]): BigInteger = using(getView)(_.getNonce(address))
+
+  override def getListOfForgerStakes: Seq[AccountForgingStakeInfo] = using(getView)(_.getListOfForgerStakes)
+
+  def getForgerStakeData(stakeId: String): Option[ForgerStakeData] = using(getView)(_.getForgerStakeData(stakeId))
+
+  override def getLogs(txHash: Array[Byte]): Array[EvmLog] = using(getView)(_.getLogs(txHash))
+
+  def getIntermediateRoot: Array[Byte] = using(getView)(_.getIntermediateRoot)
+
+  override def getCode(address: Array[Byte]): Array[Byte] = using(getView)(_.getCode(address))
+
+  override def baseFee: BigInteger = using(getView)(_.baseFee)
+
+  override def validate(tx: SidechainTypes#SCAT): Try[Unit] = Try {
+    tx.semanticValidity()
+
+    if (!tx.isInstanceOf[EthereumTransaction]) return Success()
+
```

Comment:
We should check also the chain id and reject alien txes, similarly to what we do in block validation